src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java

```diff
@@ -0,0 +1,266 @@
+  private GenomicsDBImporter importer = null;
+
+  // Intervals from command line (singleton for now)
+  private List<ChromosomeInterval> intervals;
+
+  /**
+   * From the command line validate
+   * 1. GenomicsDB workspace exists
+   * 2. GenomicsDB workspace is a valid one
+   *
+   * @return  Error messages for the exceptions
+   */
+  @Override
+  protected String[] customCommandLineValidation() {
+    String[] messages = new String[2];
+    int index = 0;
+
+    if (workspace.isEmpty()) {
+      messages[index] = " Must specify a workspace with -GW or --GenomicsDBWorkspace option";
+      return messages;
+    }
+
+    File workspaceDir = new File(workspace);
+
+    if (!createWorkspace) {
+      if (!Files.isDirectory(workspaceDir.toPath()))
+        messages[index++] = "No such directory or workspace found: " + workspace;
+
+      boolean check = new File(workspace + "/__tiledb_workspace.tdb").exists();
+
+      if (Files.isDirectory(workspaceDir.toPath()) && !check) {
+        messages[index++] = "Not a valid GenomicsDB workspace: " + workspace;
+      }
+    }
+    
+    if (sizePerColumnPartitionPerSample < 1024L) {
+      messages[index++] = "Size per column partition per sample is tool small." +
+        " Consider using larger size than 10KBytes";
+    }
+
+    return index==0 ? null : messages;
+  }
+
+  /**
+   * Before traversal, fix configuration parameters and initialize
+   * GenomicsDB. Hard-coded to handle only VCF files and headers
+   */
+  @Override
+  public void onTraversalStart() {
+
+    File workspaceDir = new File(workspace);
+
+    if (createWorkspace) {
+
+      int ret = GenomicsDBImporter.createTileDBWorkspace(workspaceDir.getAbsolutePath());
+
+      if (ret > 0) {
+        try {
+          throw new IOException("Directory " + workspaceDir + " already exists");
+        } catch (IOException e) {
+          e.printStackTrace();
+          return;
+        }
+      } else if (ret < 0) {
+        try {
+          throw new IOException("Error creating GenomicsDB workspace");
+        } catch (IOException e) {
+          e.printStackTrace();
+          return;
+        }
+      }
+    }
+
+    initializeIntervals();
+
+    List<VCFHeader> headers = new ArrayList<>();
+
+    Map<String, FeatureReader<VariantContext>> sampleToVCFMap = new HashMap<>();
+    for (String variantFile : variantFileNames) {
+      String variantFileAbsPath = new File(variantFile).getAbsolutePath();
+      AbstractFeatureReader<VariantContext, LineIterator> reader =
+        AbstractFeatureReader.getFeatureReader(variantFileAbsPath, new VCFCodec(), true);
+      headers.add((VCFHeader)reader.getHeader());
+
+      // We assume only one sample per file
+      String sampleName = ((VCFHeader) reader.getHeader()).getGenotypeSamples().get(0);
+      sampleToVCFMap.put(sampleName, reader);
+    }
+
+    Set<VCFHeaderLine> mergedHeader = VCFUtils.smartMergeHeaders(headers, true);
+    sizePerColumnPartitionPerSample *= sampleToVCFMap.size();
```

Comment:
Store the product in a different variable, instead of writing it back to `sizePerColumnPartitionPerSample`